pkg/util/discovery/discovery.go

```diff
@@ -54,3 +54,29 @@ func IsAPIGroupSupported(discoveryCli discovery.DiscoveryInterface, group string
 	}
 	return false, nil
 }
+
+// IsAPIGroupVersionSupported checks if given groupVersion and resource is supported by the cluster.
+//
+// you can exec `kubectl api-resoures` to find groupVersion and resource.
+func IsAPIGroupVersionResourceSupported(discoveryCli discovery.DiscoveryInterface, groupversion string, resource string) (bool, error) {
+	gv, err := schema.ParseGroupVersion(groupversion)
+	if err != nil {
+		return false, err
+	}
+
+	apiResourceList, err := discoveryCli.ServerResources()
```

Comment:
can we use some other methods (like GET?) and check the response to verify resource supported?

Reply:
I think it is ok, but what is the advantage?

Reply:
maybe too many `ServerResources` exist in the API server? BTW, are there some special permissions needed to call `ServerResources` (I don't know) (and `Deprecated: use ServerGroupsAndResources instead.`)?

Reply:
get it, I change to use  `ServerResourcesForGroupVersion`, i think it's more efficient.

Reply:
In my tests, I did not meet problems about permissions, so I think current permissions is ok.

Reply:
Fixed, 5e3d432559c2210687e55aca5f89ec019f2df078